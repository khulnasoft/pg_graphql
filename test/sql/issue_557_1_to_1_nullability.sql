begin;

    -- Create the party table
    create table party (
        id uuid primary key default gen_random_uuid(),
        kind varchar not null -- Indicates whether the party is a 'contact' or an 'organisation'
    );

    -- Create the contact table
    create table contact (
        id uuid primary key, -- Also a foreign key to party.id
        given_name text,
        family_name text,
        foreign key (id) references party(id)
    );

    -- Create the organisation table
    create table organization (
        id uuid primary key, -- Also a foreign key to party.id
        name text not null,
        foreign key (id) references party(id)
    );

    -- Party should have nullable relationships to Contact and Organization
    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Party") {
            kind
            fields {
              name
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }

              }
            }
          }
        }
        $$)
    );

    -- Contact and Organization should have non-nullable relationship to Party
    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Organization") {
            kind
            fields {
              name
              description
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }
              }
            }
          }
        }
        $$)
    );

    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Contact") {
            kind
            fields {
              name
              description
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }
              }
            }
          }
        }
        $$)
    );

rollback;
