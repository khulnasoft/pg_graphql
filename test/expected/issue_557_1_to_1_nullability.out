begin;
    -- Create the party table
    create table party (
        id uuid primary key default gen_random_uuid(),
        kind varchar not null -- Indicates whether the party is a 'contact' or an 'organisation'
    );
    -- Create the contact table
    create table contact (
        id uuid primary key, -- Also a foreign key to party.id
        given_name text,
        family_name text,
        foreign key (id) references party(id)
    );
    -- Create the organisation table
    create table organization (
        id uuid primary key, -- Also a foreign key to party.id
        name text not null,
        foreign key (id) references party(id)
    );
    -- Party should have nullable relationships to Contact and Organization
    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Party") {
            kind
            fields {
              name
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }

              }
            }
          }
        }
        $$)
    );
                                         jsonb_pretty                                         
----------------------------------------------------------------------------------------------
 {                                                                                           +
     "data": {                                                                               +
         "__type": {                                                                         +
             "kind": "OBJECT",                                                               +
             "fields": [                                                                     +
                 {                                                                           +
                     "name": "nodeId",                                                       +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "ID",                                                   +
                             "description": "A globally unique identifier for a given record"+
                         },                                                                  +
                         "description": null                                                 +
                     }                                                                       +
                 },                                                                          +
                 {                                                                           +
                     "name": "id",                                                           +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "UUID",                                                 +
                             "description": "A universally unique identifier"                +
                         },                                                                  +
                         "description": null                                                 +
                     }                                                                       +
                 },                                                                          +
                 {                                                                           +
                     "name": "kind",                                                         +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "String",                                               +
                             "description": "A string"                                       +
                         },                                                                  +
                         "description": null                                                 +
                     }                                                                       +
                 },                                                                          +
                 {                                                                           +
                     "name": "contact",                                                      +
                     "type": {                                                               +
                         "kind": "OBJECT",                                                   +
                         "name": "Contact",                                                  +
                         "ofType": null,                                                     +
                         "description": null                                                 +
                     }                                                                       +
                 },                                                                          +
                 {                                                                           +
                     "name": "organization",                                                 +
                     "type": {                                                               +
                         "kind": "OBJECT",                                                   +
                         "name": "Organization",                                             +
                         "ofType": null,                                                     +
                         "description": null                                                 +
                     }                                                                       +
                 }                                                                           +
             ]                                                                               +
         }                                                                                   +
     }                                                                                       +
 }
(1 row)

    -- Contact and Organization should have non-nullable relationship to Party
    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Organization") {
            kind
            fields {
              name
              description
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }
              }
            }
          }
        }
        $$)
    );
                                         jsonb_pretty                                         
----------------------------------------------------------------------------------------------
 {                                                                                           +
     "data": {                                                                               +
         "__type": {                                                                         +
             "kind": "OBJECT",                                                               +
             "fields": [                                                                     +
                 {                                                                           +
                     "name": "nodeId",                                                       +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "ID",                                                   +
                             "description": "A globally unique identifier for a given record"+
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": "Globally Unique Record Identifier"                      +
                 },                                                                          +
                 {                                                                           +
                     "name": "id",                                                           +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "UUID",                                                 +
                             "description": "A universally unique identifier"                +
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": null                                                     +
                 },                                                                          +
                 {                                                                           +
                     "name": "name",                                                         +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "String",                                               +
                             "description": "A string"                                       +
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": null                                                     +
                 },                                                                          +
                 {                                                                           +
                     "name": "party",                                                        +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "OBJECT",                                               +
                             "name": "Party",                                                +
                             "description": null                                             +
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": null                                                     +
                 }                                                                           +
             ]                                                                               +
         }                                                                                   +
     }                                                                                       +
 }
(1 row)

    select jsonb_pretty(
      graphql.resolve($$
        {
          __type(name: "Contact") {
            kind
            fields {
              name
              description
              type {
                name
                kind
                description
                ofType {
                  name
                  kind
                  description
                }
              }
            }
          }
        }
        $$)
    );
                                         jsonb_pretty                                         
----------------------------------------------------------------------------------------------
 {                                                                                           +
     "data": {                                                                               +
         "__type": {                                                                         +
             "kind": "OBJECT",                                                               +
             "fields": [                                                                     +
                 {                                                                           +
                     "name": "nodeId",                                                       +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "ID",                                                   +
                             "description": "A globally unique identifier for a given record"+
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": "Globally Unique Record Identifier"                      +
                 },                                                                          +
                 {                                                                           +
                     "name": "id",                                                           +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "SCALAR",                                               +
                             "name": "UUID",                                                 +
                             "description": "A universally unique identifier"                +
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": null                                                     +
                 },                                                                          +
                 {                                                                           +
                     "name": "givenName",                                                    +
                     "type": {                                                               +
                         "kind": "SCALAR",                                                   +
                         "name": "String",                                                   +
                         "ofType": null,                                                     +
                         "description": "A string"                                           +
                     },                                                                      +
                     "description": null                                                     +
                 },                                                                          +
                 {                                                                           +
                     "name": "familyName",                                                   +
                     "type": {                                                               +
                         "kind": "SCALAR",                                                   +
                         "name": "String",                                                   +
                         "ofType": null,                                                     +
                         "description": "A string"                                           +
                     },                                                                      +
                     "description": null                                                     +
                 },                                                                          +
                 {                                                                           +
                     "name": "party",                                                        +
                     "type": {                                                               +
                         "kind": "NON_NULL",                                                 +
                         "name": null,                                                       +
                         "ofType": {                                                         +
                             "kind": "OBJECT",                                               +
                             "name": "Party",                                                +
                             "description": null                                             +
                         },                                                                  +
                         "description": null                                                 +
                     },                                                                      +
                     "description": null                                                     +
                 }                                                                           +
             ]                                                                               +
         }                                                                                   +
     }                                                                                       +
 }
(1 row)

rollback;
